File: leo2-node/src/main/java/org/deku/leo2/node/web/ServletContextListener.java
package org.deku.leo2.node.web;

import org.apache.commons.logging.Log;
import org.apache.commons.logging.LogFactory;
import org.springframework.beans.factory.config.BeanDefinition;
import org.springframework.beans.factory.config.ConfigurableListableBeanFactory;
import org.springframework.web.context.ConfigurableWebApplicationContext;
import org.springframework.web.context.support.WebApplicationContextUtils;

import javax.inject.Named;
import javax.servlet.ServletContextEvent;

/**
 * Created by masc on 17.09.14.
 */
@Named("node.ServletContextListener")
public class ServletContextListener implements javax.servlet.ServletContextListener {
    Log mLog = LogFactory.getLog(ServletContextListener.class);

    @Override
    public void contextInitialized(ServletContextEvent sce) {
        mLog.info("Leo2 node servlet context initalizer");

        // Log all bean/names in the context
        ConfigurableWebApplicationContext wac = (ConfigurableWebApplicationContext)
                WebApplicationContextUtils.getWebApplicationContext(sce.getServletContext());

        ConfigurableListableBeanFactory clbf = wac.getBeanFactory();

        mLog.trace(String.format("Registered beans: %d", wac.getBeanDefinitionCount()));
        for (String beanName : wac.getBeanDefinitionNames()) {
            BeanDefinition rbd = clbf.getMergedBeanDefinition(beanName);
            Object s = clbf.getSingleton(beanName);
            mLog.trace(String.format("%s: %s, lazy %s",
                    beanName,
                    (s != null) ? s.getClass().getName() : "<null>",
                    rbd.isLazyInit()));
        }
    }

    @Override
    public void contextDestroyed(ServletContextEvent sce) {
        mLog.info("Leo2 servlet Context destroyed");
    }
}

File: leoz-node/src/main/java/org/deku/leoz/node/web/ServletContextListener.kt
package org.deku.leoz.node.web

import org.slf4j.LoggerFactory
import org.springframework.stereotype.Component
import org.springframework.web.context.ConfigurableWebApplicationContext
import sx.log.slf4j.trace
import javax.inject.Inject
import javax.servlet.ServletContextEvent

/**
 * Created by masc on 17.09.14.
 */
@Component("node.ServletContextListener")
class ServletContextListener : javax.servlet.ServletContextListener {
    private var log = LoggerFactory.getLogger(ServletContextListener::class.java)

    @Inject
    private lateinit var context: ConfigurableWebApplicationContext

    override fun contextInitialized(sce: ServletContextEvent) {
        log.info("Leoz servlet context initalized")

        // Log all bean/names in the context
        val clbf = context.beanFactory

        log.trace { "Registered beans: ${context.beanDefinitionCount}" }

        context.beanDefinitionNames.forEach { beanName ->
            val mbd = clbf.getMergedBeanDefinition(beanName)
            val singleton = clbf.getSingleton(beanName)
            log.trace { "${beanName}: ${singleton?.javaClass?.name ?: "<null>"}, lazy ${mbd.isLazyInit}" }
        }
    }

    override fun contextDestroyed(sce: ServletContextEvent) {
        log.info("Leoz servlet Context destroyed")
    }
}
